packages/react-button/src/components/Button/stories/ButtonIcon.stories.tsx

```diff
@@ -4,11 +4,11 @@ import { Button } from '../../../Button';
 
 export const Icon = () => (
   <>
-    <Button icon={<CalendarMonthRegular />}>Text</Button>
+    <Button icon={<CalendarMonthRegular />}>With icon before contents</Button>
     <Button icon={<CalendarMonthRegular />} iconPosition="after">
-      Text
+      With icon after contents
     </Button>
-    <Button icon={<CalendarMonthRegular />} />
+    <Button icon={<CalendarMonthRegular />} aria-label="Calendar icon" />
```

Comment:
I was a bit ambivalent on whether I should recommend changing the name to include "Calendar" just for these or for all of them. The reasoning is kind of threefold:
- The heading above the examples already sets up that these are icon examples
-  Mentioning which icon is present lets someone using a screen reader communicate with a sighted person about the examples more effectively
- It sets up developers with the pattern of using `aria-label` (or a Tooltip label) as the alt text for the icon

It's not a super strong preference though, since I think the docs will be pretty reasonable either way. I also like the idea of wrapping it in a tooltip with `relationship="label"`. It could even be a hybrid, like `content="Calendar icon only"` :)